x-pack/plugins/index_management/public/application/sections/home/index_list/index_table/index_table.js

```diff
@@ -65,17 +66,26 @@ const getColumnConfigs = ({
         defaultMessage: 'Name',
       }),
       order: 10,
-      render: (index) => (
-        <>
-          <EuiLink
-            data-test-subj="indexTableIndexNameLink"
-            onClick={() => history.push(getIndexDetailsLink(index.name, location.search || ''))}
-          >
-            {index.name}
-          </EuiLink>
-          {renderBadges(index, extensionsService, filterChanged)}
-        </>
-      ),
+      render: (index) => {
+        return (
+          <>
+            <EuiLink
+              data-test-subj="indexTableIndexNameLink"
+              onClick={() => {
+                if (!extensionsService.indexDetailsPageRoute) {
```

Comment:
would love to see some tests for both serverless and stateful

Reply:
@sabarasaba could you please guide me on best place to add these tests ?
So far I here is where I have added/updated tests: 
* Updated Jests tests to test for check for route via  extensionService - 615fb26eefecd85cc2e9838372d441ce2d08b5ac
* Stateful FTR tests to check for index_management index details page - 84d4a98284b676caca982a5abfdadb116c0da14c
* I have updated Serverless FTR tests  in this [commit](https://github.com/elastic/kibana/pull/194857/commits/0bd3ba065aab98b64d63645419ccbd756e1b3837)  but cannot tests for search_indices details page since `search_indices` is behind feature flag

Reply:
I think with the ones youve added we have enough coverage for now